workers/unity/Packages/com.improbable.gdk.tools/GenerateCode.cs

```diff
@@ -227,31 +228,21 @@ private static void CopySchema()
                 var schemaRoot = toolsConfig.SchemaSourceDirs[0];
                 CleanDestination(schemaRoot);
 
-                var packages = new Dictionary<string, string>();
-                var dependencyQueue = new Queue<string>();
-
-                // Find and include paths of all direct and nested dependencies.
-                dependencyQueue.Enqueue(Common.ManifestPath);
-                while (dependencyQueue.Count > 0)
+                // Get all packages we depend on
+                var request = Client.List(true);
+                while (!request.IsCompleted)
                 {
-                    var dependencies = GetLocalPathsInPackage(Common.ParseDependencies(dependencyQueue.Dequeue()));
-
-                    foreach (var dependency in dependencies)
-                    {
-                        if (!packages.ContainsKey(dependency.Key))
-                        {
-                            packages.Add(dependency.Key, dependency.Value);
-                            dependencyQueue.Enqueue($"{dependency.Value}/package.json");
-                        }
-                    }
+                    // Wait for the request to complete
                 }
 
-                var schemaSources = packages.Where(SchemaPathExists)
-                    .ToDictionary(kv => kv.Key, GetSchemaPath);
+                var packages = request.Result;
+                var schemaSources = packages.ToDictionary(package => package.name,
+                        package => Path.Combine(package.resolvedPath, "Schema"))
+                    .Where(kv => Directory.Exists(kv.Value));
```

Comment:
Can we do the filtering before creating the dictionary?

Reply:
That becomes tricky because I need the Schema directory, so I'd have to do that path combination twice. I think this will work fine.